File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        //tabulation
        vector<int> dp(n+1, -1);
        dp[0]=1;  // 1 way to stay at step 0
        dp[1] = 1; // 1 way to reach step 1

        for(int i=2;i<=n;i++){
            dp[i]=dp[i-1] + dp[i-2];
        }
        return dp[n];
    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        int[] dp=new int[n+1];

        return climb(n,dp);
    }
    public int climb(int n, int dp[]){
        if(n <= 2){
            return n;
        }

        if(dp[n] != 0){
            return dp[n];
        }

        dp[n] = climb(n-1, dp) + climb(n-2, dp);

        return dp[n];
    }
}